### DISCUSSION TITLE
Bind package version to assembly version

### DISCUSSION THREAD

**Discussion by forrma** [wixtoolset/issues#7517] [Questions]
My Package.wxs looks like that:

```
<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs">
    <Package Name="MyProduct" Manufacturer="MyCompany" Version="1.0.0.0" UpgradeCode="5b5b3d00-f28e-4f7b-a178-3b5dee94e0b2">
        <MajorUpgrade DowngradeErrorMessage="!(loc.DowngradeError)" />

        <Feature Id="Main">
            <ComponentGroupRef Id="ProductComponents" />
        </Feature>
    </Package>
</Wix>
```

Now I want to bind the property 'Version' (currently set to "1.0.0.0") to the AssemblyVersion of my product. I have tried different approaches, unfortunately without success.

I also tried to add the version to the msbuild, unfortunately also without success:

msbuild MyProduct.Setup.sln /property:Configuration=Release /property:SoftwareVersion=%SoftwareVersion% -p:AssemblyVersion=%SoftwareVersion% -p:Version=%SoftwareVersion% -p:FileVersion=%SoftwareVersion%

Thanks a lot.

**Comment by jhennessey** [comment]
WiX supports user defined variables that you can use for this purpose. You aren't too far off...see this stackoverflow [answer](https://stackoverflow.com/a/55349909) and the WiX documention [here](https://wixtoolset.org/docs/tools/preprocessor/#user-defined-variables). The stackoverflow page also describes how you can use binder variables if you'd prefer.
